File: Put_Marbles_in_Bags.js
/**
 * @param {number[]} weights
 * @param {number} k
 * @return {number}
 */
var putMarbles = function(weights, k) {
    let scores = [];
    for (let i = 0; i < weights.length - 1; i++) {
        scores.push(weights[i] + weights[i + 1]);
    }
    
    scores.sort((a, b) => a - b);

    let minScore = scores.slice(0, k - 1).reduce((acc, val) => acc + val, 0);
    let maxScore = scores.slice(scores.length - k + 1).reduce((acc, val) => acc + val, 0);

    return maxScore - minScore;    
};

File: Put_Marbles_in_Bags.py
class Solution:
    def putMarbles(self, weights: List[int], k: int) -> int:   
        scores = []
        for w1, w2 in zip(weights, weights[1:]):
            scores.append(w1 + w2)
        
        scores.sort()

        min_score = sum(scores[:k-1])
        max_score = sum(scores[len(scores) - k + 1:])

        return max_score - min_score
